### DISCUSSION TITLE
Compile phoenix to LLVM bitcode

### DISCUSSION THREAD

**Discussion by Elgcahlxukuth** [GothicKit/ZenKit#61] [Q&A] [answered]
Hi,

would you be able to tell whether it is possible to compile phoenix to LLVM?
maybe you already tried it? :)

cheers, Elg'

**Comment by lmichaelis** [comment, marked as answer]
Hi @Elgcahlxukuth, I do support Clang with phoenix so I believe it's just a matter of passing the correct parameters to `clang++` :>

**Reply by Elgcahlxukuth** [reply]
ok. thanks for quick replay :)
I was thinking about running some project on GraalVM, mostly in Java, but using phoenix. I'm wondering if that combination has a chance to work...

**Reply by lmichaelis** [reply]
As far as I know, the only way of interoperating Java and native code is using JNI which requires the used native library to expose a C-Interface which phoenix does not currently do.

Though I am not very familiar with GraalVM so there might be another way. If you find something, let me know!

**Reply by lmichaelis** [reply]
@Elgcahlxukuth I've been thinking about this a bit and I believe setting up a C interface is beneficial anyways. I've started work on a reduced version which you would be able to use with JNI. I'll update you when I get it ready for testing :)
